src/main/scala/viper/silver/plugin/standard/inline/InlinePredicatePlugin.scala

```diff
@@ -27,21 +27,25 @@ class InlinePredicatePlugin extends SilverPlugin with ParserPluginTemplate
 
   override def beforeVerify(input: Program): Program = {
     val rewrittenMethods = input.methods.map { method =>
-      val inlinePredIds = input.extensions.collect({case InlinePredicate(p) => p.name}).toSet
-      val recursivePreds = checkRecursive(inlinePredIds, input) ++ checkMutualRecursive(inlinePredIds, input)
-      // TODO: Do we also need to inline in inhale/exhale/assert/assume and package/apply statements?
-      val (prePredIds, postPredIds) = getPrePostPredIds(method, input, inlinePredIds)
-      val prePredIdsNoRecur = prePredIds.diff(recursivePreds.map(_.name))
-      val postPredIdsNoRecur = postPredIds.diff(recursivePreds.map(_.name))
-      val inlinedPredMethod = inlinePredicates(method, input, prePredIdsNoRecur, postPredIdsNoRecur)
-      rewriteMethod(inlinedPredMethod, input, prePredIdsNoRecur, postPredIdsNoRecur)
+      val allPredIds = input.predicates.filter(_.body.isDefined).map(_.name).toSet
+      val recursivePredIds = (checkRecursive(allPredIds, input) ++ checkMutualRecursive(allPredIds, input)).map(_.name)
```

Comment:
This is a result of me poorly-designing the `InlineErrorChecker` API, not sure why I return a list of `Predicate` instead of just their names.

Lmk if fixing this is something that'll make your downstream tasks easier.